pkg/controllers/provisioningv2/fleetcluster/fleetcluster.go

```diff
@@ -119,28 +132,33 @@ func (h *handler) createCluster(cluster *v1.Cluster, status v1.ClusterStatus) ([
 
 	agentNamespace := ""
 	clientSecret := status.ClientSecretName
+
+	objs := []runtime.Object{}
 	if mgmtCluster.Spec.Internal {
 		agentNamespace = fleetconst.ReleaseLocalNamespace
-		clientSecret = "local-cluster"
+		clientSecret = "local-kubeconfig"
 		// restore fleet's hardcoded name label for the local cluster
 		labels["name"] = "local"
-	}
-
-	var privateRepoURL string
-	if cluster.Spec.RKEConfig == nil {
-		// If the RKEConfig is nil, we are likely dealing with a legacy (v3/mgmt) cluster, and need to check the v3
-		// cluster for the cluster level registry.
-		privateRepoURL = mgmtcluster.GetPrivateRegistryURL(mgmtCluster)
-	} else {
-		privateRepoURL = image.GetPrivateRepoURLFromCluster(cluster)
+		// default cluster group, used if fleet bundle has no targets, uses hardcoded name label
+		objs = append(objs, &fleet.ClusterGroup{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "default",
```

Comment:
Is there the possibility that a user that is upgrading has not already created a `ClusterGroup` called `default` in the `fleet-local` namespace?

Do we not anticipate the potential for a conflict in name here?

`cluster.Namespace` will be `fleet-local` in this case, so we're looking at a cluster group called for `fleet-local/default`.

Reply:
Well yes. This code is replacing the [bootstrap code from Fleet](https://github.com/rancher/fleet/blob/master/pkg/controllers/bootstrap/bootstrap.go) which also ignores user modifications. It's not too bad, since wrangler's apply uses a patch and retains  most modifications.

Fleet's bootstrap would create this group unconditionally for the local cluster, too. Without this special default group every bundle would need an explicit `targets:` section.  We create this group so deployments to the local cluster behave like before. Especially since our fleet-examples repo is full of them.